client/app/bundles/course/assessment/actions.js

```diff
@@ -0,0 +1,30 @@
+import axios from 'lib/axios';
+import actionTypes from './constants';
+
+/* eslint-disable import/prefer-default-export */
+export function createAssessment(courseId, categoryId, tabId, data) {
+  const attributes = { ...data, category: categoryId, tab: tabId };
+  return (dispatch) => {
+    dispatch({ type: actionTypes.CREATE_ASSESSMENT_REQUEST });
+
+    return axios.post(`/courses/${courseId}/assessments`, attributes)
+      .then((response) => {
+        dispatch({
+          type: actionTypes.CREATE_ASSESSMENT_SUCCESS,
+          message: response.data && response.data.message,
+        });
+        // TODO: Remove redirection when assessment index is implemented using React.
+        setTimeout(() => {
+          if (response.data && response.data.id) {
+            window.location = `/courses/${courseId}/assessments/${response.data.id}`;
+          }
+        }, 200);
+      })
+      .catch((error) => {
+        dispatch({
```

Comment:
Is it better to raise `SubmissionError` so that field errors from server can be shown on the form?

Also, should the success/failure messages be kept in the frontend as much as possible since it is view-related and also since we hope to change the backend eventually?